nixos/modules/services/hardware/asusd.nix

```diff
@@ -86,19 +124,30 @@ in
       let
         maybeConfig =
           name: cfg:
-          lib.mkIf (cfg != null) {
-            source = pkgs.writeText name cfg;
-            mode = "0644";
-          };
+          lib.mkIf (cfg != null) (
+            {
+              mode = "0644";
+            }
+            // (
+              if builtins.isString cfg then
```

Comment:
IMO we can break API here, since the fix is trivial. If you really want to keep the backwards compatibility, this should at least have a warning.